automerge-protocol/src/lib.rs

```diff
@@ -303,10 +343,51 @@ impl ScalarValue {
         }
     }
 
+    pub fn to_u64(&self) -> Option<u64> {
+        match self {
+            ScalarValue::Int(n) => Some(*n as u64),
```

Comment:
Shouldn't this be stricter? I would it expect to only return a `Some` for a `Uint`, similarly for the other `to_*` functions.

Reply:
It seems like all the other `to_` functions are pretty lax?

Example:

```rust
    /// If this value can be coerced to an i64, return the i64 value
    pub fn to_i64(&self) -> Option<i64> {
        match self {
            ScalarValue::Int(n) => Some(*n),
            ScalarValue::Uint(n) => Some(*n as i64),
            ScalarValue::F32(n) => Some(*n as i64),
            ScalarValue::F64(n) => Some(*n as i64),
            ScalarValue::Counter(n) => Some(*n),
            ScalarValue::Timestamp(n) => Some(*n),
            _ => None,
        }
    }
```

Reply:
Ok, we can leave it for now then.